discovery_engine_core/tooling/src/newscatcher.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use anyhow::{Context, Result};
+use xayn_discovery_engine_providers::{Client, HeadlinesQuery, Market};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let url = "https://api-gw.xaynet.dev".to_string();
+    let token = std::env::var("GATEWAY_TOKEN").context(
+        "Please provide the GATEWAY_TOKEN environment variable for the dev environment. \
+                  The token can be found in 1Password",
+    )?;
+
+    tokio::fs::create_dir("./headlines_download")
+        .await
+        .context("Failed to create download directory. Does it already exist?")?;
+
+    let client = Client::new(token, url);
+    let market = Market {
+        lang_code: "en".to_string(),
+        country_code: "US".to_string(),
+    };
+
+    // This is updated every iteration, based on the response from Newscatcher. So in reality,
+    // we'll be fetching more than two pages.
+    let mut total_pages = 2;
+    let mut page = 1;
+    while page < total_pages {
```

Comment:
the counting is off by one, the final file and log i get is `Fetching page 99 of 100`. if the `page = total_pages` exist(?), then we should fetch it as well.
```suggestion
    // This is updated every iteration, based on the response from Newscatcher. So in reality,
    // we'll be fetching more than one page.
    let mut total_pages = 1;
    let mut page = 1;
    while page <= total_pages {
```